contrib/opencensus-ext-azure/opencensus/ext/azure/common/__init__.py

```diff
@@ -17,14 +17,73 @@
 
 from opencensus.ext.azure.common.protocol import BaseObject
 
+AUTHORIZATION = 'Authorization'
+DEFAULT_BREEZE_ENDPOINT = 'https://dc.services.visualstudio.com'
+ENDPOINT_SUFFIX = 'EndpointSuffix'
+ENV_CONNECTION_STRING = 'APPLICATIONINSIGHTS_CONNECTION_STRING'
+ENV_INSTRUMENTATION_KEY = 'APPINSIGHTS_INSTRUMENTATIONKEY'
+INGESTION_ENDPOINT = 'IngestionEndpoint'
+INSTRUMENTATION_KEY = 'InstrumentationKey'
+LOCATION = 'Location'
+
+
+def process_options(options):
+    code_cs = parse_connection_string(options.connection_string)
+    code_ikey = options.instrumentation_key
+    env_cs = parse_connection_string(os.getenv(ENV_CONNECTION_STRING))
+    env_ikey = os.getenv(ENV_INSTRUMENTATION_KEY)
+
+    options.instrumentation_key = code_cs.get(INSTRUMENTATION_KEY) \
+        or code_ikey \
+        or env_cs.get(INSTRUMENTATION_KEY) \
+        or env_ikey
+    endpoint = code_cs.get(INGESTION_ENDPOINT) \
+        or env_cs.get(INGESTION_ENDPOINT) \
+        or DEFAULT_BREEZE_ENDPOINT
+    options.endpoint = endpoint + '/v2/track'
```

Comment:
If this endpoint encoded?
Do we do canonicalization?

Reply:
Can you explain more about how the endpoint would be encoded?

Reply:
For example, do you expect the endpoint to be something like https://zh.wikipedia.org/wiki/%E6%A3%89%E5%B0%BE%E5%85%94%E5%B1%9E ?

Reply:
I do not believe that an encoded endpoint would be intentionally passed in. According to Mothra, none of the current endpoint configurations handle strings in any special way, so I believe it's safe to leave the url string as is.